File: src/lib/api/categories.ts
import { invoke } from "@tauri-apps/api/core";
import type { Category, CategoryInput } from "$lib/types";

// `includeArchived` — una categoría "eliminada" que tenía movimientos
// asociados queda archivada, no borrada (ver services::categories::delete
// en el backend). Por defecto se excluye (selección activa: chips de
// Registrar, lista de Presupuestos); pásalo en `true` donde haga falta
// resolver el nombre real de movimientos viejos (Historial, "Último
// registro"), para que no aparezcan como "Sin categoría".
export const list = (kind?: "income" | "expense", includeArchived?: boolean) =>
  invoke<Category[]>("category_list", { kind, includeArchived });

export const create = (input: CategoryInput) =>
  invoke<Category>("category_create", { input });

export const update = (id: string, name: string, isFixed: boolean, routeId: string | null) =>
  invoke<Category>("category_update", { id, name, isFixed, routeId });

export const remove = (id: string) =>
  invoke<void>("category_delete", { id });

File: src-tauri/src/commands/categories.rs
//! Comandos v2 de categorías. **No registrados en `invoke_handler!` todavía**
//! — ver nota en `commands/meta_payments.rs`.

use tauri::State;
use crate::error::AppResult;
use crate::models::{Category, CategoryInput};
use crate::services::categories as svc;
use crate::state::{DbState, get_conn};

#[tauri::command]
pub async fn category_list(state: State<'_, DbState>, kind: Option<String>) -> AppResult<Vec<Category>> {
    let conn = get_conn(&state).await?;
    svc::list(&conn, kind.as_deref()).await
}

#[tauri::command]
pub async fn category_create(state: State<'_, DbState>, input: CategoryInput) -> AppResult<Category> {
    let conn = get_conn(&state).await?;
    svc::create(&conn, input).await
}

#[tauri::command]
pub async fn category_update(
    state: State<'_, DbState>,
    id: String,
    is_fixed: bool,
    route_id: Option<String>,
) -> AppResult<Category> {
    let conn = get_conn(&state).await?;
    svc::update(&conn, &id, is_fixed, route_id).await
}

#[tauri::command]
pub async fn category_delete(state: State<'_, DbState>, id: String) -> AppResult<()> {
    let conn = get_conn(&state).await?;
    svc::delete(&conn, &id).await
}
